File: src/api/favorites.ts
// api/favorites.ts —— 收藏夹命令

import { invoke } from './client';
import type {
  FavoriteItem,
  FavoriteType,
} from '../types/favorite';

export const favoritesApi = {
  getAll: (): Promise<FavoriteItem[]> =>
    invoke<FavoriteItem[]>('favorites_get_all'),

  getByType: (favoriteType: FavoriteType): Promise<FavoriteItem[]> =>
    invoke<FavoriteItem[]>('favorites_get_by_type', { favoriteType }),

  search: (query: string): Promise<FavoriteItem[]> =>
    invoke<FavoriteItem[]>('favorites_search', { query }),

  get: (id: string): Promise<FavoriteItem | null> =>
    invoke<FavoriteItem | null>('favorites_get', { id }),

  add: (item: FavoriteItem): Promise<FavoriteItem> =>
    invoke<FavoriteItem>('favorites_add', { item }),

  update: (id: string, item: FavoriteItem): Promise<boolean> =>
    invoke<boolean>('favorites_update', { id, item }),

  remove: (id: string): Promise<boolean> =>
    invoke<boolean>('favorites_remove', { id }),

  recordUsage: (id: string): Promise<boolean> =>
    invoke<boolean>('favorites_record_usage', { id }),

  clear: (): Promise<void> => invoke<void>('favorites_clear'),

  total: (): Promise<number> => invoke<number>('favorites_total'),

  stats: (): Promise<string> =>
    invoke<string>('favorites_stats'),
};

File: src-tauri/src/commands/favorites.rs
use tauri::State;
use crate::errors::AppResult;
use crate::models::favorite::*;
use crate::services::favorites::FavoritesStore;

#[tauri::command]
pub fn favorites_get_all(store: State<'_, FavoritesStore>) -> AppResult<Vec<FavoriteItem>> { Ok(store.get_all()) }
#[tauri::command]
pub fn favorites_get_by_type(favorite_type: FavoriteType, store: State<'_, FavoritesStore>) -> AppResult<Vec<FavoriteItem>> { Ok(store.get_all().into_iter().filter(|i| i.favorite_type == favorite_type).collect()) }
#[tauri::command]
pub fn favorites_search(query: String, store: State<'_, FavoritesStore>) -> AppResult<Vec<FavoriteItem>> { Ok(store.get_all().into_iter().filter(|i| i.name.contains(&query)).collect()) }
#[tauri::command]
pub fn favorites_get(id: String, store: State<'_, FavoritesStore>) -> AppResult<Option<FavoriteItem>> { Ok(store.get(&id)) }
#[tauri::command]
pub fn favorites_add(item: FavoriteItem, store: State<'_, FavoritesStore>) -> AppResult<FavoriteItem> { Ok(store.add(item)) }
#[tauri::command]
pub fn favorites_update(id: String, item: FavoriteItem, store: State<'_, FavoritesStore>) -> AppResult<bool> { Ok(store.update(&id, item)) }
#[tauri::command]
pub fn favorites_remove(id: String, store: State<'_, FavoritesStore>) -> AppResult<bool> { Ok(store.remove(&id)) }
#[tauri::command]
pub fn favorites_record_usage(id: String, store: State<'_, FavoritesStore>) -> AppResult<bool> { Ok(store.record_usage(&id)) }
#[tauri::command]
pub fn favorites_clear(store: State<'_, FavoritesStore>) -> AppResult<()> { store.clear(); Ok(()) }
#[tauri::command]
pub fn favorites_total(store: State<'_, FavoritesStore>) -> AppResult<usize> { Ok(store.total()) }
#[tauri::command]
pub fn favorites_stats(store: State<'_, FavoritesStore>) -> AppResult<String> { Ok(format!("{} items", store.total())) }
